pipelines/test/staticpackage.yaml

```diff
@@ -0,0 +1,40 @@
+name: staticpackage
+
+# static packages that contain only static libraries, which are linked into applications at compile time rather than at runtime.
+# this packages only contains `.a` files, no other files are expected. this pipeline checks that the package contains only static libraries.
+pipeline:
+  - name: static package check
+    runs: |
+      # Get our static package name, which is usually a sub package
+      static_pkg=$(basename ${{targets.contextdir}})
+      static_files=0
+
+      # Test static library files
+      for static_file in $(apk info -qL "$static_pkg" | grep -v "^var/lib/db/sbom"); do
```

Comment:
Here is a proposal for a maybe more compact logic:

```
file_list="$(apk info --quiet --contents $static_pkg"| grep -v ^$ | grep -v var/lib/db/sbom)"

if [ "$(echo $file_list | grep .a$ | wc -l)" = 0 ] ; then
    echo "FAIL: package [$static_pkg] does not contain any static library '.a' files."
    exit 1
fi

if [  "$(echo $file_list | grep -v .a$ | wc -l)" != 0 ] ; then
   echo "FAIL: package [$static_pkg] contains unexpected files beyond the static library '.a' files."
   echo $file_list
   exit 1
fi

echo "PASS: ..."
```
This way we avoid the for loop. What do you think?